src/legacy/core_plugins/kbn_vislib_vis_types/public/components/options/point_series/point_series.tsx

```diff
@@ -100,91 +81,11 @@ function PointSeriesOptions(props: VisOptionsProps<BasicVislibParams>) {
-        <EuiTitle size="xs">
-          <h2>
-            <FormattedMessage
-              id="kbnVislibVisTypes.editors.pointSeries.thresholdLineSettings"
-              defaultMessage="Threshold Line"
-            />
-          </h2>
-        </EuiTitle>
-        <EuiSpacer size="s" />
-
-        <SwitchOption
-          label={i18n.translate('kbnVislibVisTypes.editors.pointSeries.thresholdLine.show', {
-            defaultMessage: 'Show threshold line',
-          })}
-          paramName="show"
-          value={stateParams.thresholdLine.show}
-          setValue={(paramName, value) =>
-            setValue('thresholdLine', { ...stateParams.thresholdLine, [paramName]: value })
-          }
-        />
-
-        {stateParams.thresholdLine.show && (
-          <>
-            <NumberInputOption
-              label={i18n.translate(
-                'kbnVislibVisTypes.editors.pointSeries.thresholdLine.valueLabel',
-                {
-                  defaultMessage: 'Threshold value',
-                }
-              )}
-              paramName="value"
-              value={stateParams.thresholdLine.value}
-              setValue={(paramName, value) =>
-                setValue('thresholdLine', { ...stateParams.thresholdLine, [paramName]: value || 0 })
-              }
-            />
-
-            <NumberInputOption
-              label={i18n.translate(
-                'kbnVislibVisTypes.editors.pointSeries.thresholdLine.widthLabel',
-                {
-                  defaultMessage: 'Line width',
-                }
-              )}
-              paramName="width"
-              min={1}
-              step={1}
-              value={stateParams.thresholdLine.width}
-              setValue={(paramName, value) =>
-                setValue('thresholdLine', { ...stateParams.thresholdLine, [paramName]: value || 1 })
-              }
-            />
-
-            <SelectOption
-              label={i18n.translate('kbnVislibVisTypes.editors.pointSeries.thresholdLine.style', {
-                defaultMessage: 'Line style',
-              })}
-              options={options}
-              paramName="style"
-              value={stateParams.thresholdLine.style}
-              setValue={(paramName, value) =>
-                setValue('thresholdLine', { ...stateParams.thresholdLine, [paramName]: value })
-              }
-            />
-
-            <EuiFormRow
-              label={i18n.translate('kbnVislibVisTypes.editors.pointSeries.thresholdLine.color', {
-                defaultMessage: 'Line color',
-              })}
-            >
-              <EuiColorPicker
-                color={stateParams.thresholdLine.color}
-                onChange={value => {
-                  setValue('thresholdLine', { ...stateParams.thresholdLine, color: value });
-                }}
-              />
-            </EuiFormRow>
-          </>
-        )}
-      </EuiPanel>
+      {stateParams.thresholdLine && <ThresholdPanel {...props} />}
```

Comment:
@flash1293 I added this check because not all vis have a config for `thresholdLine` like `Horizontal bar` (currently `Panel setting` is broken on master for `Horizontal bar`).  Does this behavior correct? 
I guess that `thresholdLine` config wasn't added for `Horizontal bar` just by mistake. If it is I'll add the config. 
Or maybe there should be default values for `ThresholdPanel` in case of the `thresholdLine` config absence in vis...

Reply:
Good question, I think you are right - in `src/legacy/ui/public/vislib/visualizations/point_series/_point_series.js` line 117ff it looks like there is code handling the horizontal case, so it should work and be shown in the UI.

As there is a `show` parameter already I would go with adding it to `Horizontal bar` initial config instead of having default values in place - ping @ppisljar

Reply:
Thank you. I added the config to `Horizontal bar`.